provision-osx/tasks/configure_buildfarm_proxy.yml

```diff
@@ -0,0 +1,22 @@
+---
+
+-
+  name: Set node proxy configuration with authentication
+  shell: "networksetup -{{ item }} {{ proxy_device }} {{ proxy_host }} {{ proxy_port }} on {{ proxy_user }} {{ proxy_pass }}"
+  with_items: "{{ proxy_ctx }}"
+  become: true
+  when: proxy_host is defined and proxy_user is defined and proxy_pass is defined
+
+-
+  name: Set node proxy configuration without authentication
+  shell: "networksetup -{{ item }} {{ proxy_device }} {{ proxy_host }} {{ proxy_port }}"
+  with_items: "{{proxy_ctx}}"
+  become: true
+  when: proxy_host is defined and proxy_user is not defined and proxy_pass is not defined
```

Comment:
We should also add a check here for empty values. So if the proxy_user and proxy_pass are uncommented but have empty values we should also run the one without authentication. There is an example of this [here](https://github.com/fheng/rhmap-ansible/blob/master/roles/proxy-setup/tasks/main.yml#L16-L20)